tutorials/stats-sensor-space/40_cluster_1samp_time_freq.py

```diff
@@ -86,60 +98,125 @@
+# dimension matching the data size (excluding observations) in an
+# "unrolled" format, so: len(channels × frequencies × times)
 assert adjacency.shape[0] == adjacency.shape[1] == \
     len(tfr_epochs.ch_names) * len(tfr_epochs.freqs) * len(tfr_epochs.times)
 
 # %%
 # Compute statistic
 # -----------------
-threshold = 3.
-n_permutations = 50  # Warning: 50 is way too small for real-world analysis.
+# For forming clusters, we need to specify a critical test statistic threshold.
+# Only data bins exceeding this threshold will be used to form clusters.
+#
+# Here, we
+# use a t-test and can make use of Scipy's percent point function of the t
+# distribution to get a t-value that corresponds to a specific alpha level
+# for significance. This threshold is often called the
+# "cluster forming threshold".
+#
+# .. note::
+#    The choice of the threshold is more or less arbitrary. Choosing
+#    a t-value corresponding to p=0.05, p=0.01, or p=0.001 may often provide
+#    a good starting point. Depending on the specific dataset you are working
+#    with, you may need to adjust the threshold.
+
+# We want a two-tailed test
+tail = 0
+
+# In this example, we wish to set the threshold for including data bins in
+# the cluster forming process to the t-value corresponding to p=0.001 for the
+# given data.
+#
+# Because we conduct a two-tailed test, we divide the p-value by 2 (which means
+# we're making use of both tails of the distribution).
+# As the degrees of freedom, we specify the number of observations
+# (here: epochs) minus 1.
+# Finally, we subtract 0.001 / 2 from 1, to get the critical t-value
+# on the right tail (this is needed for MNE-Python internals)
+degrees_of_freedom = len(epochs) - 1
+t_thresh = scipy.stats.t.ppf(1 - 0.001 / 2, df=degrees_of_freedom)
+
+# Set the number of permutations to run.
+# Warning: 50 is way too small for a real-world analysis (where values of 5000
+# or higher are used), but here we use it to increase the computation speed.
+n_permutations = 50
+
+# Run the analysis
 T_obs, clusters, cluster_p_values, H0 = \
     permutation_cluster_1samp_test(epochs_power, n_permutations=n_permutations,
-                                   threshold=threshold, tail=0,
+                                   threshold=t_thresh, tail=tail,
                                    adjacency=adjacency,
                                    out_type='mask', verbose=True)
 
 # %%
 # View time-frequency plots
 # -------------------------
-
+# We now visualize the observed clusters that are statistically significant
+# under our permutation distribution.
+#
+# .. warning:: Talking about "significant clusters" can be convenient, but
+#              you must be aware of all associated caveats! For example, it
+#              is **invalid** to assume that a cluster at channels A and B
+#              over 0ms to 100ms means that there is a difference at these
+#              particular channels and time points. See the comprehensive
```

Comment:
I'm not sure about this wording. There is a univariate difference (uncorrected) as assessed by the stat_fun, it's just that the cluster p value does not relate to a null hypothesis that is so spatio-temporally specific. Maybe something like this:
```suggestion
#              is **invalid** to interpret the cluster p value as being
#              spatially or temporally specific. A cluster with sufficiently
#              low (for example < 0.05) p value at specific location does not
#              allow you to say that the significant effect is at that
#              particular location. The p value only tells you about the
#              probability of obtaining similar or stronger/larger cluster
#              anywhere in the data if there were no differences between the
#              compared conditions. So it only allows you to draw conclusions
#              about the differences in the data "in general", not at specific
#              locations. See the comprehensive
#              `FieldTrip tutorial <ft_cluster_>`_ for more information.
```
I'm sure this will be confusing for begginners however we phrase that, because it is counterintuitive, but maybe this explanation will help to point in the right direction.